Formula/pcre.rb

```diff
@@ -43,8 +48,7 @@ def install
       --enable-pcregrep-libbz2
     ]
 
-    # JIT not currently supported for Apple Silicon
-    args << "--enable-jit" unless Hardware::CPU.arm?
+    args << "--enable-jit" if want_jit?
```

Comment:
```suggestion
    # JIT not currently supported for Apple Silicon
    args << "--enable-jit" if MacOS.version >= :high_sierra && Hardware::CPU.intel?
```

Reply:
The logic in the suggested change isn't equivalent, since it uses `:high_sierra` rather than `:sierra` as a boundary, and "not arm" is not necessarily the same as "intel".

Reply:
Misread the macOS version check. However, I'm pretty sure `Hardware::CPU` recognises only ARM, Intel, or PPC. And I suspect most of `brew` doesn't run on PPC, so `!Hardware::CPU.arm?` may as well be `Hardware::CPU.intel?`.

Reply:
I can fire up my G4 running snow leopard to check :)  Probably faster to try running against emulation via QEMU.

Regardless, I saw in the history that somebody had specifically changed away from an explicit check for intel to a check for "not arm" (in 16678a015ed73d8e0e1c2de1ae75e3b5052253df).  The commit log offered no insight into why, so I simply tried to honor that change.

Reply:
That's fine. We can do `!Hardware::CPU.arm?`.

Reply:
I amended the commit and force-pushed the change, but it looks like github doesn't reflect that into this conversation.

Most of my day-to-day work is under gitlab, which renders these sorts of things differently.  If you'd prefer that the PR have two distinct commits (the original, and a commit for the review results), I'm happy to rework.

Reply:
No, what you have here is fine now.